bindings/go/src/fdb/tuple/tuple.go

```diff
@@ -106,109 +115,128 @@ func adjustFloatBytes(b []byte, encode bool) {
 	}
 }
 
-func encodeBytes(buf *bytes.Buffer, code byte, b []byte) {
-	buf.WriteByte(code)
-	buf.Write(bytes.Replace(b, []byte{0x00}, []byte{0x00, 0xFF}, -1))
-	buf.WriteByte(0x00)
+type packer struct {
+	buf []byte
```

Comment:
What's benefit of wrapping the byte slice in a struct (rather than just keeping a raw slice and calling append on it as necessary)? Is it just to have a handle wrapping the slice to make it mutable (or mutable in length)?

There was a question on the forums about what it would take to make golang support versionstamps (https://forums.foundationdb.org/t/golang-versionstamp-usage/253). I think the answer might be "add a field to the packer struct saying at what position you wrote a versionstamp that will be completed at commit time", so we might want it anyway.

Reply:
I went with a struct because it makes the code more ergonomic (i.e. we can afford to have a `p.putBytes` method because it always gets inlined). Looking at the version stamps issue, it seems like it's a good choice rather than passing around all the data through the call stack.